src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -382,6 +383,13 @@ private int FindEntry(TKey key)
                         }
 
                         i = entries[i].next;
+                        if (loops >= entries.Length)
+                        {
+                            // The chain of entires forms a loop; which means a concurrent update has happened.
+                            // Break out of the loop and throw; rather than looping forever.
+                            ThrowHelper.ThrowInvalidOperationException_ConcurrentOperationsNotSupported();
+                        }
```

Comment:
What kind of measurable impact, if any, does this have on TryGetValue?

Reply:
It shouldn't impact items that haven't collided (as they don't loop); hoping it will be minimal; but will have a look.

On the flip side, what's the alternative?

Reply:
> On the flip side, what's the alternative?

If it's very negatively impactful, not doing it.